app/controllers/checkouts_controller.rb

```diff
@@ -7,4 +7,12 @@ def create
 
     redirect_to root_path
   end
+
+  def destroy
+    @book = Book.find(params[:book_id])
+    @checkout = Checkout.where(user: current_user, book: @book)
+    @checkout.destroy_all
```

Comment:
I think you can accomplish this all in one line - or at least 1 statement:

```ruby
Checkout.
  where(user_id: current_user, book_id: params[:book_id]).
  destroy_all
```

I don't think you need to assign any instance variables here as we're not redering them in any views.